### PR TITLE
Upgrade Python version to 3.11

### PR DESCRIPTION
**Pull request by tenzen-y**
**What this PR does / why we need it**:
I upgraded the Python version to 3.11 since Python 3.10 no longer fixes bugs, according to [this](https://devguide.python.org/versions/#supported-versions).

Also, I upgraded some libraries to switch Python 3.11 supported version:

- numpy 1.25.2: https://numpy.org/neps/nep-0029-deprecation_policy.html#support-table

Additionally, we keep the library and Python version for the skopt since some dependencies error occurs due to stopped maintenance: #2280.

**Which issue(s) this PR fixes** _(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)_:
Fixes #

**Checklist:**

- [ ] [Docs](https://www.kubeflow.org/docs/components/katib/) included if any changes are user facing

### CODE REVIEW

**Comment by tenzen-y** [comment]
/hold

We need to upgrade other library versions first.

File: manifests/v1beta1/installs/katib-standalone/katib-config.yaml [hunk 1]
```diff
@@ -45,7 +45,7 @@ runtime:
       image: docker.io/kubeflowkatib/suggestion-enas:latest
       resources:
         limits:
-          memory: 200Mi
+          memory: 400Mi
```
Comment (tenzen-y) [inline manifests/v1beta1/installs/katib-standalone/katib-config.yaml:48]: After I upgraded the Python version, I faced OOM killed in the ENAS suggestion service.

**Comment by tenzen-y** [comment]
Ready for the review. PTAL.

/assign @andreyvelich @johnugeorge

**Review by andreyvelich — COMMENTED** [review]
Thank you @tenzen-y!
A few questions:
1. Do we want to add support for Python 3.11 in Katib SDK: https://github.com/kubeflow/katib/blob/master/sdk/python/v1beta1/setup.py#L61 ?
2. Do we want to add this change before the next Katib release ?

**Comment by tenzen-y** [comment]
> Thank you @tenzen-y! A few questions:
> 
> 1. Do we want to add support for Python 3.11 in Katib SDK: https://github.com/kubeflow/katib/blob/master/sdk/python/v1beta1/setup.py#L61 ?
> 2. Do we want to add this change before the next Katib release ?

@andreyvelich It's a good question. 

1. I agree with you. If so, should we remove Python 3.7 from the list as Python 3.7 is no longer supported? 
2. Adding support in this PR would be better since we can verify if our SDK with Python 3.11 can work for a while by adding CI: https://github.com/kubeflow/katib/blob/679e6fb8f8768b4d23d640ea56103a20fdb027fe/.github/workflows/e2e-test-darts-cifar10.yaml#L24

**Comment by andreyvelich** [comment]
>I agree with you. If so, should we remove Python 3.7 from the list as Python 3.7 is no longer supported?

How strict we want to be with our SDK versions since it is user facing interface and sometimes users might be on the out-of-date environments ?
For example, Kubernetes Python client still supports Python 3.6 and 3.7: https://github.com/kubernetes-client/python/blob/master/setup.py#L80.

However, I prefer approach which [Jupyter Lab](https://github.com/jupyterlab/jupyterlab/blob/main/pyproject.toml#L13) and [Jupyter Server](https://github.com/jupyter-server/jupyter_server/blob/main/pyproject.toml#L24) follow where they add minimal Python version that is supported by today.

Could you give your feedback on this @kubeflow/wg-training-leads @droctothorpe @shipengcheng1230 @andreafehrman @harrisonfritz

**Comment by tenzen-y** [comment]
> How strict we want to be with our SDK versions since it is user facing interface and sometimes users might be on the out-of-date environments ? For example, Kubernetes Python client still supports Python 3.6 and 3.7: https://github.com/kubernetes-client/python/blob/master/setup.py#L80.
> 
> However, I prefer approach which [Jupyter Lab](https://github.com/jupyterlab/jupyterlab/blob/main/pyproject.toml#L13) and [Jupyter Server](https://github.com/jupyter-server/jupyter_server/blob/main/pyproject.toml#L24) follow where they add minimal Python version that is supported by today.

I'm ok with either way, but I believe that we should define deprecation or support policies for SDK with Python version.
For example, we decided to support 3 Kubernets minor versions.

So, should we keep supported Python versions in this PR, and then open a dedicated issue and discuss the Python version?
This means that I should not add Python 3.11 to SDK and should not remove Python 3.7 from SDK.

@andreyvelich WDYt?

**Comment by andreyvelich** [comment]
>So, should we keep supported Python versions in this PR, and then open a dedicated issue and discuss the Python version?
This means that I should not add Python 3.11 to SDK and should not remove Python 3.7 from SDK.

That's fine I believe. Let's create dedicated issue to discuss long-term support for SDK Python versions.

**Comment by tenzen-y** [comment]
> > So, should we keep supported Python versions in this PR, and then open a dedicated issue and discuss the Python version?
> > This means that I should not add Python 3.11 to SDK and should not remove Python 3.7 from SDK.
> 
> That's fine I believe. Let's create dedicated issue to discuss long-term support for SDK Python versions.

I created: #2283

**Comment by tenzen-y** [comment]
@andreyvelich So, can we merge this?

**Comment by andreyvelich** [comment]
Yeah, let's merge it!
/lgtm

**Pull request merged**